concurrency: include promotion information when printing out locks

arulajmani opened the pull request:
We only do so if a request is promoting its locks; otherwise, the lock formatting remains the same as before (which is nice becuase it reduces useless test churn).

Followup from https://github.com/cockroachdb/cockroach/pull/119671

Release note: None

Epic: none

arulajmani committed:
concurrency: include promotion information when printing out locks

We only do so if a request is promoting its locks; otherwise, the lock
formatting remains the same as before (which is nice becuase it reduces
useless test churn).

Followup from https://github.com/cockroachdb/cockroach/pull/119671

Release note: None

Epic: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120918)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/120918#-:-Nti0dwbEODGQxQHZ1BB:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @arulajmani)

___
*[`pkg/kv/kvserver/concurrency/lock_table.go` line 1893 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/120918#-Nti-_sZ2yzGePuyhbpp:-Nti-_sZ2yzGePuyhbpq:bdm3507) ([raw file](https://github.com/cockroachdb/cockroach/blob/ae4be2225a2b1068b6bc314584e3dcda19105643/pkg/kv/kvserver/concurrency/lock_table.go#L1893)):*
> ```Go
> 			}
> 			sb.Printf("    active: %t req: %d%s, strength: %s, txn: ",
> 				redact.Safe(qg.active), redact.Safe(qg.order.reqSeqNum), redact.Safe(optPromotingMsg),
> ```

Are any of these `redact.Safe` calls necessary? `optPromotingMsg` is currently because it's a `string` and not a `redact.SafeString`, but we can fix that with `optPromotingMsg := redact.SafeString("")`. Otherwise, the other arguments are a `bool`, a `uint64`, and a lock.Strength (which implements `redact.SafeValue`). So I don't think any of them need `redact.Safe` wrapping. Mind checking?

By the way, I'm asking both to avoid verbosity but also because `redact.Safe` has a runtime cost.


<!-- Sent from Reviewable.io -->

arulajmani committed:
concurrency: rename tests

These referenced lockState, which has since been renamed to keyLocks.

Epic: none

Release note: None

arulajmani reviewed (commented):
TFTR! 

bors r=nvanbenschoten

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/120918#-:-Nul6odA5WNXjyh91MNQ:b-ieq29z)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @nvanbenschoten)

___
*[`pkg/kv/kvserver/concurrency/lock_table.go` line 1893 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/120918#-Nti-_sZ2yzGePuyhbpp:-Nul6UOnDUeWNsuq9nc_:b-1wz1vj) ([raw file](https://github.com/cockroachdb/cockroach/blob/ae4be2225a2b1068b6bc314584e3dcda19105643/pkg/kv/kvserver/concurrency/lock_table.go#L1893)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Are any of these `redact.Safe` calls necessary? `optPromotingMsg` is currently because it's a `string` and not a `redact.SafeString`, but we can fix that with `optPromotingMsg := redact.SafeString("")`. Otherwise, the other arguments are a `bool`, a `uint64`, and a lock.Strength (which implements `redact.SafeValue`). So I don't think any of them need `redact.Safe` wrapping. Mind checking?

By the way, I'm asking both to avoid verbosity but also because `redact.Safe` has a runtime cost.
</blockquote></details>

Yeah, you're right, these were superfluous. Removed, and took your suggestion for `optPromotingMsg`. 

I also added the formatting test I used to verify.  


<!-- Sent from Reviewable.io -->
